indico/modules/notifications/util.py

```diff
@@ -0,0 +1,70 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2019 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from __future__ import unicode_literals
+
+from functools import wraps
+
+from indico.core.db import db
+from indico.core.db.sqlalchemy.descriptions import RenderMode
+from indico.core.notifications import email_sender
+from indico.modules.notifications import settings as notification_settings
+from indico.modules.notifications.models.notifications import Notification
+from indico.util.caching import memoize
+
+
+@memoize
```

Comment:
Not a good idea, that way changing the setting won't ever be taking into account unless the server is restarted. `memoize_request` would work, but settings are already cached during the request so no need to memoize anything here.

Reply:
Indeed. Good point.